### ISSUE TITLE
Multiple Wallets within one instance

### ISSUE THREAD

**Issue by renepickhardt** [ElementsProject/lightning#1897]
Looking at https://github.com/shesek/spark-wallet/blob/master/doc/dev-regtest-env.md I see that it is currently possible to run two lightning wallets (maybe I should say nodes) on one machine. 

But I guess there is quite some overhead running the deamon let's say 100 times instead of running the deamon 1 time and allowing 100 data dirs and expose each dir on a different port. 

Is it planned to inlcude this as a feature any time soon?

**Comment by ZmnSCPxj** [comment]
I think (but do not know) much of the overhead is running multiple processes, one for each open channel.  Presumably 100 wallets would have 100 times more channels than a single wallet. I suspect any savings for using a single `lightningd` instance will be lost in that we create a `channeld`/`openingd`/`closingd`/`onchaind` for each open channel.  Perhaps the only true savings would be `gossipd`, since it would be reasonable to share the map across wallets.

**Comment by rustyrussell** [comment]
I don't think this is worthwhile, since it introduces issues with restoration, which is critical.
